view.tag.php

```diff
@@ -12,163 +12,165 @@
 /**
  * XOOPS tag management module
  *
+ * @package         XoopsModules\Tag
  * @copyright       XOOPS Project (https://xoops.org)
  * @license         http://www.fsf.org/copyleft/gpl.html GNU public license
  * @since           1.00
  * @author          Taiwen Jiang <phppp@users.sourceforge.net>
  * */
 
-use Xmf\Request;
 use XoopsModules\Tag\Constants;
+use XoopsModules\Tag\Utility;
+use XoopsModules\Tag\Common;
 
 require_once __DIR__ . '/header.php';
 
-//xoops_loadLanguage('main', 'tag');
-/*
-if (!($GLOBALS["xoopsModule"] instanceof XoopsModule) || "tag" != $GLOBALS["xoopsModule"]->getVar("dirname", "n")) {
-    xoops_loadLanguage("main", "tag");
+//@todo refactor this code - it "works" but it's not right. Look at previous revs using $args_num to see what it's suppose to do
+if (Utility::tag_parse_args($args, $args_string)) {
+    $args['tag']  = !empty($args['tag']) ? $args['tag'] : (!empty($args_string[0]) ? $args_string[0] : Constants::DEFAULT_ID);
```

Comment:
In use, looking at a tag "bbb":
Tag links are built as *modules/tag/view.tag.php?bbb*
but, following link results in a message: "Invalid query"

The "Jump to" box works and will get to "bbb" tagged items

It seems to work by changing line 30 in view.tag.php to:
```
    $args['tag']  = !empty($args['tag']) ? $args['tag'] : (is_numeric($args_string[0]) ? $args_string[0] : Constants::DEFAULT_ID);
//                                                         ^^^^^^^^^^
```
Without the change both $args['tag'] and $args['term'] get set to "bbb" and it results in an "Invalid query" message.

Reply:
Thanks! I've included your fix